sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TimeWindow.scala

```diff
@@ -110,10 +115,26 @@ object TimeWindow {
    *         precision.
    */
   def getIntervalInMicroSeconds(interval: String): Long = {
-    val cal = IntervalUtils.stringToInterval(UTF8String.fromString(interval))
-    if (cal.months != 0) {
-      throw new IllegalArgumentException(
-        s"Intervals greater than a month is not supported ($interval).")
+    val ymIntervalErrMsg = s"Intervals greater than a month is not supported ($interval)."
+    val cal = try {
+      if (interval.toLowerCase(Locale.ROOT).trim.startsWith("interval")) {
+        CatalystSqlParser.parseExpression(interval) match {
+          case Literal(_: Int, _: YearMonthIntervalType) =>
+            throw new IllegalArgumentException(ymIntervalErrMsg)
+          case Literal(micros: Long, _: DayTimeIntervalType) =>
+            new CalendarInterval(0, 0, micros)
+        }
+      } else {
+        val parsed = IntervalUtils.stringToInterval(UTF8String.fromString(interval))
+        if (parsed.months != 0) {
+          throw new IllegalArgumentException(ymIntervalErrMsg)
+        } else {
+          parsed
+        }
+      }
+    } catch {
+      case NonFatal(e) =>
+        throw QueryCompilationErrors.cannotParseTimeDelayError(interval, e)
     }
     cal.days * MICROS_PER_DAY + cal.microseconds
```

Comment:
Not related to the PR but:
1. `*` and `+` can overflow. I think we should use exact arithmetic ops here.
2. one day in `CalendarInterval` can be 23, 24, or 25 hours, see https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java#L42

Reply:
O.K, let's fix this in this PR.

Reply:
For `1`, I've fixed in this PR.
For `2`, it's about intervals, not timestamps so we don't need to consider daylight saving here right?

Reply:
> For 2, it's about intervals, not timestamps so we don't need to consider daylight saving here right?

ok. Let's keep the assumption of 24 hours per day here. Not sure it will work fine during daylight saving time but it seems we don't have enough test coverage for that at the moment.

Maybe, open an JIRA to test the case?

Reply:
Hmm, I'll check it out and open a JIRA.